Description: Fix race condition causing build failures on i386.
Origin: http://git.debian.org/?p=pkg-multimedia/lame.git;a=blob;f=debian/patches/parallel-builds-fix.patch
Forwarded: yes
Applied-Upstream: http://lame.cvs.sf.net/viewvc/lame/lame/libmp3lame/i386/Makefile.am?revision=1.27

--- a/libmp3lame/i386/Makefile.am
+++ b/libmp3lame/i386/Makefile.am
@@ -15,6 +15,7 @@
 if HAVE_NASM
 noinst_LTLIBRARIES = liblameasmroutines.la
 liblameasmroutines_la_SOURCES = $(nasm_sources)
+liblameasmroutines_la_DEPENDENCIES = $(nasm_sources:.nas.lo)
 am_liblameasmroutines_la_OBJECTS = \
 	choose_table$U.lo \
 	cpu_feat$U.lo \
@@ -53,11 +54,10 @@
 	$(NASM) $(NASMFLAGS) $< -o $@ -l $@.lst
 
 .nas.lo: $< nasm.h
-	mkdir -p .libs
 	$(ECHO) '# Generated by ltmain.sh - GNU libtool 1.5.22 (1.1220.2.365 2005/12/18 22:14:06)' >$@
-	$(ECHO) "pic_object='.libs/$*.o'" >>$@
-	$(ECHO) "non_pic_object='.libs/$*.o'" >>$@
-	$(NASM) $(NASMFLAGS) $< -o .libs/$*.o -l $@.lst
+	$(ECHO) "pic_object='$*.o'" >>$@
+	$(ECHO) "non_pic_object='$*.o'" >>$@
+	$(NASM) $(NASMFLAGS) $< -o $*.o -l $@.lst
 
 COMPILE = $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) \
 	$(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
